src/gp_doctest.cpp

```diff
@@ -191,5 +208,111 @@ TEST_CASE("GPInstance: Hotstart branch lengths") {
       0.1892030000, 0.1894900000, 0.1895430000, 0.1896900000, 0.1905710000;
   double true_mean = hotstart_expected_branch_lengths.array().mean();
   inst.HotStartBranchLengths();
+  std::cout << inst.GetEngine()->GetBranchLengths() << std::endl;
+  std::cout << true_mean << std::endl;
   CHECK_EQ(true_mean, inst.GetEngine()->GetBranchLengths()(2));
 }
+
+GPInstance MakeFiveTaxaInstance() {
+  GPInstance inst("_ignore/mmapped_plv.data");
+  inst.ReadFastaFile("data/five_taxon.fasta");
+  inst.ReadNewickFile("data/five_taxon_rooted.nwk");
+  inst.MakeEngine();
+  return inst;
+}
+
+TEST_CASE("GPInstance: generate all trees") {
+  auto inst = MakeFiveTaxaInstance();
+  auto rooted_tree_collection = inst.GenerateCompleteRootedTreeCollection();
+  CHECK_EQ(rooted_tree_collection.TreeCount(), 4);
+  CHECK_EQ(rooted_tree_collection.TopologyCounter().size(), 4);
+}
+
+TEST_CASE("GPInstance: marginal likelihood on five taxa") {
+  auto inst = MakeFiveTaxaInstance();
+  inst.EstimateBranchLengths(1e-6, 10);
+  double gp_marginal_log_likelihood = inst.GetEngine()->GetLogMarginalLikelihood();
+  double exact_marginal_log_likelihood = ComputeExactMarginal(
+      "data/five_taxon_unrooted_with_branch_lengths.nwk", "data/five_taxon.fasta");
+  CHECK_LT(abs(exact_marginal_log_likelihood - gp_marginal_log_likelihood), 1e-6);
+}
+
+TEST_CASE("GPInstance: test populate PLV") {
+  // Estimate branch lengths. This will populate and update PLVs. Then compute
+  // likelihoods. PopulatePLV, which reset log marginal as well as zeros out PLVs before
+  // re-populating them using the current branch lengths. Compute likelihoods. Compare
+  // the likelihood vectors.
+  auto inst = MakeFiveTaxaInstance();
+  inst.EstimateBranchLengths(1e-6, 10);
+  inst.ComputeLikelihoods();
+  const EigenVectorXd log_likelihoods1 = inst.GetEngine()->GetLogLikelihoods();
+  inst.PopulatePLVs();
+  inst.ComputeLikelihoods();
+  const EigenVectorXd log_likelihoods2 = inst.GetEngine()->GetLogLikelihoods();
+  CheckVectorXdEquality(log_likelihoods1, log_likelihoods2, 1e-6);
+}
+
+TEST_CASE("GPInstance: SBN probabilities on five taxa") {
```

Comment:
I don't want to diminish what you've done, but it seems to me that this only tests SBN probabilities at the rootsplits, right? If so, let's just add "rootsplit probabilities" or someting to the name here.

Reply:
Done.